open-api/rest-catalog-open-api.yaml

```diff
@@ -2643,7 +2643,7 @@ components:
       properties:
         type:
           type: string
-          enum: [ "assert-ref-snapshot-id" ]
+          enum: ["tag", "branch"]
```

Comment:
I don't think this change is correct. In https://github.com/apache/iceberg/blob/d69ba0568a2e07dfb5af233350ad5668d9aef134/core/src/main/java/org/apache/iceberg/UpdateRequirementParser.java#L169 we're relying on the `type` being `assert-ref-snapshot-id`

Reply:
I think the type duplicates the one in the parent entity. @anuragmantri made a fix here, https://github.com/apache/iceberg/pull/10434